apis/Google.LongRunning/Google.LongRunning/Operation.cs

```diff
@@ -105,17 +129,37 @@ public Operation(Operation rpcMessage, OperationsClient client)
         /// Retrieves the metadata associated with this operation, or <c>null</c> if there is no
         /// metadata in the underlying response message.
         /// </summary>
-        public TMetadata Metadata => RpcMessage.Metadata?.Unpack<TMetadata>();
+        /// <remarks>
+        /// Only the in-memory representation of the operation (this object) is consulted for its state.
+        /// See the <see cref="Operation"/> documentation for information about dealing with different metadata type versions.
+        /// </remarks>
+        /// <exception cref="InvalidOperationException">Metadata is present, but is not of the expected type.</exception>
```

Comment:
Use a custom exception `IncompatibleMetadataException` (or something), for clarity?